File: packages/web/src/enums/groceryCategory.ts
export enum GroceryCategory {
  PRODUCE = "produce",
  DAIRY = "dairy", 
  MEAT = "meat",
  FROZEN = "frozen",
  BAKERY = "bakery",
  PANTRY = "pantry",
  BEVERAGES = "beverages",
  HOUSEHOLD = "household",
  OTHER = "other",
}

export const GroceryCategoryLabelMap = {
  [GroceryCategory.PRODUCE]: "Fruits & Vegetables",
  [GroceryCategory.DAIRY]: "Dairy",
  [GroceryCategory.MEAT]: "Meat & Poultry", 
  [GroceryCategory.FROZEN]: "Frozen Foods",
  [GroceryCategory.BAKERY]: "Bakery & Grains",
  [GroceryCategory.PANTRY]: "Pantry & Grains",
  [GroceryCategory.BEVERAGES]: "Beverages",
  [GroceryCategory.HOUSEHOLD]: "Household",
  [GroceryCategory.OTHER]: "Other",
}

export enum GroceryViewMode {
  CATEGORIZED = "categorized",
  ALPHABETICAL = "alphabetical", 
  UNCATEGORIZED = "uncategorized",
}

export const GroceryViewModeLabelMap = {
  [GroceryViewMode.CATEGORIZED]: "Categorized",
  [GroceryViewMode.ALPHABETICAL]: "Alphabetical",
  [GroceryViewMode.UNCATEGORIZED]: "List View",
}

export const CategoryOrder = [
  GroceryCategory.PRODUCE,
  GroceryCategory.DAIRY,
  GroceryCategory.MEAT,
  GroceryCategory.FROZEN,
  GroceryCategory.BAKERY,
  GroceryCategory.PANTRY,
  GroceryCategory.BEVERAGES,
  GroceryCategory.HOUSEHOLD,
  GroceryCategory.OTHER,
]

File: packages/ios/Kairos/Kairos/Enums/GroceryCategory.swift
import Foundation

/// Mirrors `packages/web/src/enums/groceryCategory.ts` (web-only).
enum GroceryCategory: String, Codable, CaseIterable, Sendable {
    case produce
    case dairy
    case meat
    case frozen
    case bakery
    case pantry
    case beverages
    case household
    case other

    var displayName: String {
        switch self {
        case .produce: "Fruits & Vegetables"
        case .dairy: "Dairy"
        case .meat: "Meat & Poultry"
        case .frozen: "Frozen Foods"
        case .bakery: "Bakery & Grains"
        case .pantry: "Pantry & Grains"
        case .beverages: "Beverages"
        case .household: "Household"
        case .other: "Other"
        }
    }

    static let displayOrder: [GroceryCategory] = [
        .produce, .dairy, .meat, .frozen, .bakery, .pantry, .beverages, .household, .other
    ]
}
